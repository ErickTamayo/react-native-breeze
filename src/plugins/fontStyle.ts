import { PluginFunction } from "./types";

export type PluginGroups = { style: "italic" | "normal" };

export const pattern = /^(?<style>italic|not-italic)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { style } = groups;
  return { fontStyle: style === "italic" ? "italic" : "normal" };
};
