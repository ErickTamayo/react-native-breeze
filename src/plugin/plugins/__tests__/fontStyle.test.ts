import {
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../fontStyle";

describe("fontStyle", () => {
  const shouldMatch = ["italic", "not-italic"];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
