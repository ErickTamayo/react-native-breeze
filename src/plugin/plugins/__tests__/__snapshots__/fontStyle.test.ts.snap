// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fontStyle Should parse (italic) correctly 1`] = `
Object {
  "fontStyle": "italic",
}
`;

exports[`fontStyle Should parse (not-italic) correctly 1`] = `
Object {
  "fontStyle": "normal",
}
`;
